src/main/cpp/include/config/pb_config.h

```diff
@@ -0,0 +1,41 @@
+/**
+ * The MIT License (MIT)
+ * Copyright (c) 2019 Omics Data Automation, Inc
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef PB_CONFIG_H
+#define PB_CONFIG_H
+
+#include "genomicsdb_config_base.h"
+#include "genomicsdb_export_config.pb.h"
+
+class PBConfigBase : public GenomicsDBConfigBase {
```

Comment:
This is probably a dumb question, but why is this GenomicsDBConfigBase here instead of VariantQueryConfig?

Reply:
Not a dumb question  by any means - I have struggled to determine what's the best class hierarchy here (or whether it's even needed)
```suggestion
A<--B  B is a subclass of A
GenomicsDBConfigBase <-- JSONConfigBase
GenomicsDBConfigBase <-- PBConfigBase
GenomicsDBConfigBase <-- GenomicsDBImportConfig
GenomicsDBConfigBase <-- VariantQueryConfig
```
So, you could create a PBConfigBase object and then:
```
  PBConfigBase pb_cfg;
  //initializes struct here and then ..
  VariantQueryConfig qcfg;
  static_cast<GenomicsDBConfigBase&>(qcfh) = std::move(static_cast<GenomicsDBConfigBase&>(pb_cfg));
```
An alternative is to get rid of JSONConfigBase and PBConfigBase and move the functions to GenomicsDBConfigBase (read_from_JSON_file, read_from_PB). Would probably simplify matters.

Reply:
> static_cast<GenomicsDBConfigBase&>(qcfh) = std::move(static_cast<GenomicsDBConfigBase&>(pb_cfg));
>An alternative is to get rid of JSONConfigBase and PBConfigBase and move the functions to GenomicsDBConfigBase (read_from_JSON_file, read_from_PB). 
Yes, this would simplify things as reading from JSON/protobuf files would be part of the base configuration, e.g. specifying workspace would be treated the same whether it came from GenomicsDBImportConfig api directly or JSON files or protobuf. It would also avoid moving JSON|PBConfigBase properties back into VariantQueryConfig in a roundabout way.

Reply:
One other question not related to this PR and to just increase my understanding, why is the bookkeeping operation not part of instantiating VariantQueryProcessor? Should it remain separate or can we merge it in?

Reply:
The bookkeeping depends on the input query (JSON/Protobuf) of the user (fields, bounds etc).